test/cpp/test_tstr.cpp

```diff
@@ -4,6 +4,35 @@
 
 #include "microtest.h"
 
+TEST(TStrFmt) {
```

Comment:
Can you add test that is on the limit, e.g. 10*1024 caracters, case with +1 and -1 char?